### PR TITLE
PluggableUSB support

### PR DESCRIPTION
**Pull request by bjc**
Adds support for PluggableUSB API.

### CODE REVIEW

**Comment by maxgerhardt** [comment]
Given my original comment at the PlatformIO build script still needs to be adapted for 

```
 build.usb_flags=-DUSB_VID={build.vid} -DUSB_PID={build.pid} "-DUSB_MANUFACTURER={build.usb_manufacturer}" "-DUSB_PRODUCT={build.usb_product}"
```
otherwise compilation will fail, akin to something like in [stm32duino](https://github.com/stm32duino/Arduino_Core_STM32/blob/main/tools/platformio/platformio-build.py#L98-L112).

Another point of issue is guarding the compilation + usage of USB functions for only those boards / chip series that support it, otherwise compilation for other boards will break. I'm e.g. seeing a rather unconditional include of USB-stack headers in `cores/arduino/USBCore.h` and an unconditional init of the USB stick in `cores/arduino/main.cpp` which concerns me.

@bjc do you as the original author want to have a look at refactoring / guarding this or should I? Either works for me. I'm eager to getting this merged before I do any major core refactorings.

**Comment by obra** [comment]
@maxgerhardt I think that if you've got the cycles, it's worth you taking a look at doing that guarding. (You also have more experience with our other GD32 platforms)

**Comment by maxgerhardt** [comment]
Good to merge after some adaptions made in https://github.com/CommunityGD32Cores/ArduinoCore-GD32/tree/merge_usb.

File: cores/arduino/api/PluggableUSB.cpp [hunk 1]
```diff
@@ -17,85 +17,105 @@
   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
-#include "USBAPI.h"
```
Comment (maxgerhardt) [inline cores/arduino/api/PluggableUSB.cpp:20]: Modification of supposedly constant ArduinoCore-API files is not good, but I guess needed here. Flagging for the future.

**Pull request merged**